wgengine/netstack/netstack.go

```diff
@@ -1533,7 +1533,7 @@ func (ns *Impl) forwardUDP(client *gonet.UDPConn, clientAddr, dstAddr netip.Addr
 		ns.logf("could not get backend local IP:port from %v:%v", backendLocalAddr.IP, backendLocalAddr.Port)
 	}
 	if isLocal {
-		ns.pm.RegisterIPPortIdentity(backendLocalIPPort, dstAddr.Addr())
+		ns.pm.RegisterIPPortIdentity("udp", backendLocalIPPort, clientAddr.Addr())
```

Comment:
So, this is an actual bug; we were registering the identity of the destination (i.e. the local `tailscaled` node) instead of the client. There are, unfortunately, no tests for this; I discovered it when testing manually. I'll look into adding some.